src/dawn_native/vulkan/TextureVk.cpp

```diff
@@ -258,6 +256,17 @@ namespace dawn_native { namespace vulkan {
         createInfo.pQueueFamilyIndices = nullptr;
         createInfo.initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
 
+        switch (GetDimension()) {
+            case dawn::TextureDimension::e2D:
+                createInfo.extent = VkExtent3D{GetWidth(), GetHeight(), 1};
+                createInfo.arrayLayers = GetDepth();
```

Comment:
[1]: I don't think that depth will be for both 3D texture depth and the 2D array size. Vulkan and Metal have it separate while only D3D12 has the combined `DepthOrArraySize` in the texture descriptors. Can a new "SetArraySlice" be added to the TextureBuilder, or even better could the texture be descriptorized? It is something we need to do to get closer to WebGPU anyway.

Reply:
Got it. I will descriptorize texture first, then support 2D array textures. The implementation of "BufferDescriptor" should be a good reference for me.

Reply:
Thanks!

Reply:
Done